core/app/api/v0/c/[communitySlug]/site/[...ts-rest]/route.ts

```diff
@@ -794,6 +588,97 @@ const handler = createNextHandler(
 				};
 			},
 		},
+		forms: {
+			getPubsForFormField: async ({ params, query }, { responseHeaders }) => {
+				const { user, community, authorization } = await checkAuthorization({
+					token: { scope: ApiAccessScope.pub, type: ApiAccessType.read },
+					cookies: "community-member",
+				});
+
+				let { pubTypeId, stageId, pubIds, filters, ...rest } = query ?? {};
+
+				if (query?.filters) {
+					try {
+						await validateFilter(community.id, query.filters);
+					} catch (e) {
+						throw new BadRequestError(e.message);
+					}
+				}
+
+				const [form, userCanAccessForm] = await Promise.all([
+					getForm({
+						slug: params.formSlug,
+						communityId: community.id,
+					}).executeTakeFirst(),
+					userHasAccessToForm({
+						userId: user.id,
+						communityId: community.id,
+						formSlug: params.formSlug,
+						pubId: params.pubId,
+					}),
+				]);
+
+				if (!form) {
+					throw new NotFoundError();
+				}
+
+				if (!userCanAccessForm) {
+					throw new ForbiddenError();
+				}
+
+				// check if user has access to the form
+				const field = form.elements.find((e) => e.slug === params.fieldSlug);
+
+				if (
+					!field ||
+					field.type !== ElementType.pubfield ||
+					field.isRelation !== true ||
+					field.config.relationshipConfig.component !== InputComponent.relationBlock
+				) {
+					throw new NotFoundError(
+						`Field ${params.fieldSlug} not found on form ${params.formSlug}`
+					);
+				}
+
+				const formFieldPubTypes = field.relatedPubTypes;
+
+				if (formFieldPubTypes.length === 0) {
+					throw new BadRequestError(
+						`Field ${params.fieldSlug} on form ${params.formSlug} does not allow any pub types`
+					);
+				}
+
+				const [pubs, pubCount] = await Promise.all([
+					getPubsWithRelatedValues(
+						{
+							communityId: community.id,
+							pubTypeId: pubTypeId,
+							stageId: stageId,
+							pubIds: pubIds,
+							// userId: user.id,
+						},
+						{
+							...rest,
+							filters: query?.filters,
+							allowedPubTypes: formFieldPubTypes,
+						}
+					),
+					getPubsCount({
+						communityId: community.id,
+						pubTypeId: pubTypeId,
+						stageId: stageId,
+					}),
+				]);
+
+				// TODO: this does not account for permissions
+				responseHeaders.set(TOTAL_PUBS_COUNT_HEADER, `${pubCount}`);
+
+				return {
+					status: 200,
+					body: pubs,
+				};
+			},
+		},
```

Comment:
this does practically the same as `pub.getMany`, but 
- checks whether the user has access to the form
- only gets the pubs that match the pubtype of that field on the form
- does not take into account the users permissions